addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/service/SmappeeDeviceReading.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.smappee.service;
+
+/**
+ * The result of a smappee reading
+ *
+ * @author Niko Tanghe
+ */
+
+public class SmappeeDeviceReading {
+    public int serviceLocationId;
+    public SmappeeDeviceReadingConsumption[] consumptions;
+
+    public double getLatestConsumption() {
+        if (consumptions.length == 0) {
+            return 0;
+        }
+        return getLatestReading().consumption;
+    }
+
+    public double getLatestSolar() {
+        if (consumptions.length == 0) {
+            return 0;
+        }
+        return getLatestReading().solar;
+    }
+
+    public double getLatestAlwaysOn() {
+        if (consumptions.length == 0) {
+            return 0;
+        }
+        return getLatestReading().alwaysOn;
+    }
+
+    private SmappeeDeviceReadingConsumption getLatestReading() {
+        if (consumptions.length == 0) {
```

Comment:
This check doesn't add anything because all callers already check for it. And they should because without that check a null would be returned causing a nullpointer as it's not checked for being null in the callers of this method.

Reply:
sometimes the api gives me an empty array for the readings, it's because smappee groups the readings a bit for a specific time interval, the array itself is never null (as it should be for a good api)
so the check should be there on all methods, except for getLatestReading, returning null is not ok here, removed the test)